src/dbnode/storage/shard.go

```diff
@@ -974,6 +962,62 @@ func (s *dbShard) writeAndIndex(
 	return series, wasWritten, nil
 }
 
+func (s *dbShard) SeriesReadWriteRef(
+	id ident.ID,
+	tags ident.TagIterator,
+	opts ShardSeriesReadWriteRefOptions,
+) (SeriesReadWriteRef, error) {
+	// Try retrieve existing series.
+	entry, shardOpts, err := s.tryRetrieveWritableSeries(id)
```

Comment:
Does this need a lock btw? Or is access here entirely singlethreaded?

Reply:
Thankfully `tryRetrieveWritableSeries` will lock.